packages/sources/ice/src/transport/price.ts

```diff
@@ -0,0 +1,141 @@
+import {
+  EndpointContext,
+  LwbaResponseDataFields,
+} from '@chainlink/external-adapter-framework/adapter'
+import { TransportDependencies } from '@chainlink/external-adapter-framework/transports'
+import {
+  StreamingTransport,
+  SubscriptionDeltas,
+} from '@chainlink/external-adapter-framework/transports/abstract/streaming'
+import {
+  makeLogger,
+  ResponseGenerics,
+  sleep,
+  TimestampedAdapterResponse,
+} from '@chainlink/external-adapter-framework/util'
+import { BaseEndpointTypes } from '../endpoint/price'
+import { LocalPriceCache } from './cache'
+import { InstrumentPartialUpdate, PartialPriceUpdate, StreamingClient } from './netdania'
+
+const logger = makeLogger('ice: StreamingTransport')
+
+export type FullPriceUpdate = Required<PartialPriceUpdate> & {
+  firstTs: number
+  version: number // incremented on each update; useful to disambiguate updates with the same ts
+}
+
+export class NetDaniaStreamingTransport extends StreamingTransport<BaseEndpointTypes> {
+  private client!: StreamingClient
+  private localCache: LocalPriceCache = new LocalPriceCache()
+
+  async initialize(
+    dependencies: TransportDependencies<BaseEndpointTypes>,
+    adapterSettings: BaseEndpointTypes['Settings'],
+    endpointName: string,
+    transportName: string,
+  ) {
+    await super.initialize(dependencies, adapterSettings, endpointName, transportName)
+    this.client = new StreamingClient(adapterSettings)
+    this.client.on('price', async (update: InstrumentPartialUpdate) => {
+      // get base and quote from the instrument name in the requestIdToInstrument map
+      const base = update.instrument.substring(0, 3)
```

Comment:
this limits to 3 chars tickers only

Reply:
Yes. All of them are like so. Maybe we should add a custom config validation rule instead.